tests/test_datasets/test_dataset_functions.py

```diff
@@ -963,7 +960,7 @@ def test_create_dataset_pandas(self):
         )
         self.assertEqual(
             _get_online_dataset_format(dataset.id),
-            'sparse_arff',
+            'arff',
```

Comment:
Shouldn't this still be sparse arff? I guess something changed internally based on the fact that the dataframe is now a regular dataframe, but I think this shouldn't be the case in practice.

Reply:
Based on the Pandas update this PR addresses, the dataframe `df` created on _L931_ is a standard Pandas data frame. However, `df.dtypes` will reveal that each column is instead a sparse array. Maybe since the outer abstraction is no longer a SparseDataFrame but a standard DataFrame, it is being stored as arff and not sparse-arff? Since I had to make that change for the test case to get it to work.

Reply:
@mfeurer I changed the check inside `create_dataset` to detect a sparse data frame based on the new pandas check. However, that for the concerned test case here, it falters at `publish()` with an `OpenMLServerException: Error parsing dataset ARFF file - Arff error in dataset file: trailing characters on data line  (l.9)`